src/java.base/share/classes/sun/reflect/generics/factory/CoreReflectionFactory.java

```diff
@@ -89,6 +90,26 @@ public static CoreReflectionFactory make(GenericDeclaration d, Scope s) {
         return new CoreReflectionFactory(d, s);
     }
 
+    public static Type ownerType(Class<?> c) {
```

Comment:
This feels somewhat odd. How about, moving these back to Executable and rename `ownerType` to `resolveToType` ? I realise we can't get use getFactory() in executable but as you write, it isn't needed, it should be fine in this case to create it directly as in your v1. The reason is the semantics of this is somewhat unintuitive and is only used in the annotated receiver case.

Reply:
I moved the code back and it's fairly compact now. I added "owner" to the method name since it does not make sense otherwise. Let me know what you think!